components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -5447,6 +5447,29 @@ public List<RoleV2> getAssociatedRolesOfApplication(String applicationId, String
         }
     }
 
+    @Override
+    public void addAssociatedRoleToApplication(String applicationUUID, String roleId)
+            throws IdentityApplicationManagementException {
+
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(true)) {
+
+            try (PreparedStatement statement = connection.prepareStatement(
+                    ApplicationMgtDBQueries.ADD_APPLICATION_ASSOC_ROLE)) {
```

Comment:
shall we use both connection and prepared statement as resources in side one try

Reply:
fixed